Fortran/exercise2/league_sim_mod.f90

```diff
@@ -0,0 +1,85 @@
+! module for league simulation routines
+module league_simulator
+
+implicit none
+
+integer :: max_goals = 5
+
+private
+
+public :: simulate_match, sort_teams, max_goals
+
+contains
+
+! simulates a match between two teams
+subroutine simulate_match(team1, team2, points)
+    implicit none
+    integer, intent(in) :: team1, team2
+    integer, intent(inout) :: points(:)
+    
+    integer :: goals1, goals2
+    real :: g1, g2        
+
+    call random_seed()
+
+    ! Simulate goals for each team (max 5 goals)
+    call random_number(g1)
+    call random_number(g2)
+    goals1 = floor(max_goals*g1) 
+    goals2 = floor(max_goals*g2)
+    
+    ! Update points based on match result
+    if (goals1 > goals2) then
+        points(team1) = points(team1) + 3   ! Team 1 wins
+    else if (goals2 > goals1) then
+        points(team2) = points(team2) + 3   ! Team 2 wins
+    else
+        points(team1) = points(team1) + 1   ! Draw, each team gets 1 point
+        points(team2) = points(team2) + 1
+    end if
+   
+end subroutine simulate_match
+
+! sorts an array of teams by the points in array points
```

Comment:
```suggestion
! sort the teams by points scored
```
Suggest this more informative comment. Remember, comments shouldn't just tell the reader what the code below is doing, it is more useful if they help describe why that block of code is being run in the context of the overall module.